crates/matrix-sdk-crypto/src/olm/account.rs

```diff
@@ -1305,12 +1305,22 @@ impl Account {
                 // we might try to create the same session again.
                 // TODO: separate the session cache from the storage so we only add
                 // it to the cache but don't store it.
-                store
-                    .save_changes(Changes {
-                        sessions: vec![result.session.clone()],
-                        ..Default::default()
-                    })
-                    .await?;
+                let mut changes =
+                    Changes { sessions: vec![result.session.clone()], ..Default::default() };
+                // Any new Olm session will bump the Olm wedging index for the
+                // sender's device, if we have their device, which will cause
+                // us to re-send existing Megolm sessions to them the next time
+                // we use the session.  If we don't have their device, this
+                // means that we haven't tried to send them any Megolm sessions
+                // yet, so we don't need to worry about it.
+                if let Some(device) = store.get_device_from_curve_key(sender, sender_key).await? {
+                    let mut read_only_device = device.inner;
+                    read_only_device.olm_wedging_index += 1;
```

Comment:
Technically this can panic, what do you think about using our `SequenceNumber` type for this. Not that it's terribly important but since we have such a type, seems like we could just use it for correctness sake:
https://github.com/matrix-org/matrix-rust-sdk/blob/3be84a5a3014abc2bdd3018392d6a43c92c01b1d/crates/matrix-sdk-crypto/src/store/caches.rs#L191-L229

You would need to derive `Serialize` and `Deserialize` for it with the [`transparent`](https://serde.rs/container-attrs.html#transparent) attribute, but that's about it.